packages/x-date-pickers/src/internals/hooks/useField/useFieldV7TextField.ts

```diff
@@ -113,7 +113,11 @@ export const useFieldV7TextField: UseFieldTextField<true> = (params) => {
         return sectionListRef.current.getSectionIndexFromDOMElement(activeElement);
       },
       focusField: (newSelectedSections = 0) => {
-        if (!sectionListRef.current) {
+        if (
+          !sectionListRef.current ||
+          // if the field is already focused, we don't need to focus it again
```

Comment:
Avoid resetting/jumping focus to the first section when switching fields with the mouse on a multi input range field.

https://github.com/user-attachments/assets/1482e65e-a3e1-450b-b4e1-bab7fd980f70